src/Tests/Tests.Reproduce/GithubIssue3356.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using Elastic.Xunit.XunitPlumbing;
+using Elasticsearch.Net;
+using FluentAssertions;
+using Nest;
+using Nest.JsonNetSerializer;
+
+namespace Tests.Reproduce
+{
+	public class GithubIssue3356
+	{
+		[U]
+		public void JoinFieldDeserializedCorrectly()
+		{
+			var pool = new SingleNodeConnectionPool(new Uri("http://localhost:9200"));
```

Comment:
LGTM small suggestion though is to use the recently centralized json serialization assertions.

```csharp
var tester = SerializationTester.DefaultWithSourceSerializer;
var response = tester.Client.IndexDocument(doc);
tester.AssertSerialize(response.ApiCall.RequestBodyInBytes, new { join = "parent" });
var doc = tester.AssertDeserialize<MyDocument>(response.ApiCall.RequestBodyInBytes);
```

Reply:
`SerializationTester.DefaultWithSourceSerializer;` 

Would need to be added though

Reply:
or even `AssertRoundTrip<MyDocument>()`